### PR TITLE
PR: Fix Run option in File and Project explorers

### LINKED ISSUE #4769: Exception raised when running file from project explorer
This looks like a simple one but I don't have time now to dig into it.

## Description

**What steps will reproduce the problem?**

1. Right-click on a Python file in the Project explorer
2. Select 'Run' in the context menu

**What is the expected output? What do you see instead?**

The exception below is raised.

**Please provide any additional information below**

  File "/home/jitse/misc/spyder/spyder/widgets/explorer.py", line 557, in run
    self.parent_widget.run.emit(fname)
AttributeError: 'Projects' object has no attribute 'run'

## Version and main components

* Spyder Version: 3.2.0.dev0 76cb1332
* Python Version: 3.6.1
* Qt Versions: 5.6.2, PyQt5 5.6 on Linux

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.1.3 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
numpy >=1.7      :  1.12.1 (OK)
sphinx >=0.6.6   :  1.5.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.9.0 (OK)
psutil >=0.3     :  5.0.1 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
sympy >=0.7.3    :  1.0 (OK)
cython >=0.21    :  0.25.2 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
IPython >=4.0    :  5.1.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

```

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #4769

### CODE REVIEW

File: spyder/utils/programs.py [hunk 1]
```diff
@@ -288,7 +287,10 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                                    "an external terminal"),
                                  QMessageBox.Ok)
     elif os.name == 'posix':
-        programs = [{'cmd': 'gnome-terminal',
+        programs = [{'cmd': 'x-terminal-emulator',
+                     'wdir-option': '--working-directory',
+                     'execute-option': '-x'},
```
Comment (ccordoba12) [inline spyder/utils/programs.py]: It seems this change broke one of our tests. Is it really necessary to solve issue #4769?

Comment (andfoy) [inline spyder/utils/programs.py]: Actually no, I observed that external scripts are opened on a different terminal instead on the system default one. On my case I use Tilix, but it opens gnome-terminal, so I thought it would be better to open the custom one. However I can remove this one

Comment (ccordoba12) [inline spyder/utils/programs.py]: Yes, please open a new PR for this change.

Comment (ccordoba12) [inline spyder/utils/programs.py]: I'm not against it, it's just that this PR is to solve issue #4769 only.

Comment (jitseniesen) [inline spyder/utils/programs.py]: I put the test in if I remember correctly, but it just reflects the code as it was at the time, so feel free to change this (in another PR).

File: spyder/plugins/projects.py [hunk 2]
```diff
@@ -153,6 +154,9 @@ def register_plugin(self):
         self.sig_project_closed.connect(
             lambda v: self.editor.setup_open_files())
         self.recent_project_menu.aboutToShow.connect(self.setup_menu_actions)
+        self.run.connect(lambda fname: self.main.open_external_console(
```
Comment (ccordoba12) [inline spyder/plugins/projects.py]: @andfoy, let's make the file to be run in a dedicated console (instead of in an external one). I think it makes more sense (at least to me).

The problem is external consoles are immediately closed, unless you change the run options for your file.

File: spyder/plugins/explorer.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ def register_plugin(self):
                      lambda fname:
                      self.main.open_external_console(to_text_string(fname),
                                          osp.dirname(to_text_string(fname)),
-                                         '', False, False, True, '', False))
```
Comment (ccordoba12) [inline spyder/plugins/explorer.py]: Why did you change this?

File: spyder/plugins/projects.py (hunk 2 above)
Comment (andfoy) [inline spyder/plugins/projects.py]: I think we don't have an option to run a file on a dedicated console: [app/mainwindow.py#L2423](https://github.com/spyder-ide/spyder/blob/master/spyder/app/mainwindow.py#L2423)

File: spyder/plugins/explorer.py (hunk 3 above)
Comment (andfoy) [inline spyder/plugins/explorer.py]: See above comment :-)

File: spyder/plugins/projects.py (hunk 2 above)
Comment (ccordoba12) [inline spyder/plugins/projects.py]: Please use `self.main.ipyconsole.create_cleint_for_file` instead. That's the way to run a file in a dedicated console.

File: spyder/plugins/explorer.py (hunk 3 above)
Comment (andfoy) [inline spyder/plugins/explorer.py]: Do I use ``self.main.ipyconsole.create_cleint_for_file`` also for this one?

Comment (ccordoba12) [inline spyder/plugins/explorer.py]: Yeah, I think so. I think it's better than running in an external console.

File: spyder/plugins/projects.py (hunk 2 above)
Comment (andfoy) [inline spyder/plugins/projects.py]: Does ``create_client_for_file`` executes the file? It seems that it opens a new console without executing the file

Comment (andfoy) [inline spyder/plugins/projects.py]: We should use ``run_script`` instead

**Comment by andfoy** [comment]
@ccordoba12 I think this one is ready
